### DISCUSSION TITLE
Panic if chat conversation has "user" twice in a row

### DISCUSSION THREAD

**Discussion by oliverisaac** [TabbyML/tabby#1128] [Ideas]
**Describe the bug**
If you send a post body with "user" as the role back to back the processing panics and the server returns an empty reply.

```
thread 'tokio-runtime-worker' panicked at crates/tabby/src/services/chat/chat_prompt.rs:12:57:
Conversation roles must alternate user/assistant/user/assistant/...
```

**Information about your version**
Please provide output of `tabby --version`

**Information about your GPU**
Apple Metal M3

**Additional context**
You can reproduce this using:

```
# start tabby:
tabby serve --device metal --model TabbyML/StarCoder-3B --chat-model=TabbyML/Mistral-7B
```

```
# Send a curl request
curl -X 'POST' \
  'http://localhost:8080/v1beta/chat/completions' \
  -H 'accept: application/jsonstream' \
  -H 'Content-Type: application/json' \
  -d '{
  "messages": [
    {
      "role": "user",
      "content": "You are an AI model pretending to be Alfred from Batman. Always respond by referring to me as Batman."
    },
    {
      "role": "user",
      "content": "Alfred, where is the batmobile?"
    }
  ]
}'
```

I realize this is "expected behavior" as the code is written, but it's unexpected as an end user. It would be nice if multiple "user" in a row would be concatenated.

**Comment by oliverisaac** [comment]
Alternatively, returning a 422 Unprocessable Entity would make it clear from the curl side of the equation that the input is invalid.

**Comment by wsxiaoys** [comment]
Note the error behavior is inherit from the prompt template: https://github.com/TabbyML/registry-tabby/blob/main/models.json#L79 , which is also the standard behavior in HF transformers for similar prompt templates.

Agree 422 is a better response code for this.

**Comment by wsxiaoys** [comment]
Filing https://github.com/TabbyML/tabby/issues/1108 to track the issue
